File: erpnext/accounts/report/cash_and_bank_status/cash_and_bank_status.js
// Copyright (c) 2016, Frappe Technologies Pvt. Ltd. and contributors
// For license information, please see license.txt

frappe.query_reports["Cash and Bank Status"] = {
    "filters": [
		{
			"fieldname":"company",
			"label": __("Company"),
			"fieldtype": "Link",
			"options": "Company",
			"default": frappe.defaults.get_user_default("Company"),
			"reqd": 1
		},
        {
            "fieldname":"from_date",
            "label": __("From Date"),
            "fieldtype": "Date",
			'reqd': 1,
            "default": frappe.datetime.add_days(frappe.datetime.nowdate(), -30)
        },
        {
            "fieldname":"to_date",
            "label": __("To Date"),
            "fieldtype": "Date",
			'reqd': 1,
            "default":frappe.datetime.nowdate()
        },
    ],
	get_chart_data: function(columns, result) {
		return {
			data: {
				columns: [
					['إجمالى الوارد'].concat($.map(result, function(d) { return d[0]; })),
					['إجمالى المنصرف'].concat($.map(result, function(d) { return d[1]; }))
				]
				// rows: [['Date', 'Mins to first response']].concat(result)
			},
			chart_type: 'bar',
			bar: {
				width: 150
				//		{
				//	ratio: 0.5
				//}
				//	ratio: 0.5 // this makes bar width 50% of length between ticks
				// or
				//width: 100 // this makes bar width 100px
			}
		}
	}
}

File: erpnext/accounts/report/cash_and_bank_status/cash_and_bank_status.py
# Copyright (c) 2013, Frappe Technologies Pvt. Ltd. and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe

def execute(filters=None):
	columns = [
		{
			'fieldname': 'إجمالى الوارد',
			'label': 'إجمالى الوارد',
			'fieldtype': 'Currency',
			"width": 500
		},
		{
			'fieldname': 'إجمالى المنصرف',
			'label': 'إجمالى المنصرف',
			'fieldtype': 'Currency',
			"width": 500
		},
	]

	data = frappe.db.sql("""select sum(income) as 'إجمالى الوارد', sum(expences) as 'إجمالى المنصرف' from (
					select sum(g.debit) as income, sum(g.credit) as expences from `tabGL Entry` g 
					left join `tabAccount` ac on ac.name=g.account 
					left join `tabAccount` ag on ag.name=g.against 
					where g.posting_date>=%(from_date)s and g.posting_date<=%(to_date)s
					and g.company=%(company)s 
					and ifnull(ac.account_type,'') in ('Cash','Bank') 
					and not ifnull(ag.account_type,'') in ('Cash','Bank')
					) a
					""", {'from_date': filters.from_date, 'to_date': filters.to_date, 'company': filters.company})

	return columns, data
